### PR TITLE
Modify server.start() so server can continue to respond to requests

### PR DESCRIPTION
**Pull request by GeorgeSmith-Sweeper**

### CODE REVIEW

File: src/main/java/com/GeorgesServer/app/RequestHandler.java [hunk 1]
```diff
@@ -4,16 +4,17 @@
...
             return clientRequest.get("HttpVersion") + " 200 OK" + "\n";
         }
 
-        // SIMPLE GET
         if (clientRequest.get("Method").equals("GET") && clientRequest.get("Url").equals("/")) {
             return clientRequest.get("HttpVersion") + " 200 OK" + "\n";
         }
-        return clientRequest.get("HttpVersion") + " 200 OK" + "\n";
 
+        return defaultResponse;
```
Comment (GeorgeSmith-Sweeper) [inline src/main/java/com/GeorgesServer/app/RequestHandler.java]: This line would pass both simple post and simple get on it's own.

File: src/main/java/com/GeorgesServer/app/EstablishesConnection.java [hunk 2]
```diff
@@ -3,25 +3,29 @@
...
+            Socket clientSocket = this.serverSocket.accept();
+            InputStream inputStream = clientSocket.getInputStream();
+            OutputStream outputStream = clientSocket.getOutputStream();
+            InputStreamReader inReader = new InputStreamReader(inputStream);
+            OutputStreamWriter outWriter = new OutputStreamWriter(outputStream);
+            BufferedReader buffReader = new BufferedReader(inReader);
+            return new Connections(buffReader, outWriter);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
```
Comment (nicole-a-tesla) [inline src/main/java/com/GeorgesServer/app/EstablishesConnection.java:28]: There are a few of these in here. Not the end of the world just now, but when you notice null checks spreading through your code it's time for an abstraction

File: src/main/java/com/GeorgesServer/app/MyServer.java [hunk 3]
```diff
@@ -8,18 +8,25 @@
...
     }
 
-    public void start(int port) {
-        Connections connections = establishesConnection.connect(port);
-        HashMap<String, String> clientRequest = requestParser.parse(connections.getIn());
-        String serverResponse = requestHandler.handle(clientRequest);
-        responseSender.send(serverResponse, connections.getOut());
+    public void start() {
+        String serverResponse = "";
+        while (!serverResponse.equals("Bye")) {
```
Comment (nicole-a-tesla) [inline src/main/java/com/GeorgesServer/app/MyServer.java:25]: 😆

File: src/main/java/com/GeorgesServer/app/RequestHandler.java [hunk 4]
```diff
@@ -4,9 +4,17 @@
 
 public class RequestHandler {
     public String handle(HashMap<String, String> clientRequest) {
+
+        String defaultResponse = clientRequest.get("HttpVersion") + " 200 OK" + "\n";
```
Comment (nicole-a-tesla) [inline src/main/java/com/GeorgesServer/app/RequestHandler.java:8]: Lots of duplication of ` + " 200 OK" + "\n";` in here.

File: src/main/java/com/GeorgesServer/app/RequestHandler.java [hunk 5]
```diff
@@ -4,9 +4,17 @@
...
+
+        if (clientRequest.get("Method").equals("POST") && clientRequest.get("Url").equals("/form")) {
```
Comment (nicole-a-tesla) [inline src/main/java/com/GeorgesServer/app/RequestHandler.java:10]: Kind of a smell to have two identical returns for different conditionals, but you're pretty early in this process so :shrug: Clean this up as it grows.

File: src/main/java/com/GeorgesServer/app/RequestReader.java [hunk 6]
```diff
@@ -7,8 +7,9 @@
 
     public String read(BufferedReader reader) {
         StringBuilder sb = new StringBuilder();
+        String line;
         try {
-            sb.append(reader.readLine());
+            if ((line = reader.readLine()) != null) { sb.append(line); }
         } catch (IOException e) {
             e.printStackTrace();
             return null;
```
Comment (nicole-a-tesla) [inline src/main/java/com/GeorgesServer/app/RequestReader.java:15]: Keep an eye on these nulls

**Comment by nicole-a-tesla** [comment]
Looking good overall!

**Pull request merged**